wr executor

sb10 opened the pull request:
Work in progress for an executor that uses wr as a scheduler. See https://github.com/nextflow-io/nextflow/issues/1088

pditommaso commented on modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy:
```diff
@@ -0,0 +1,664 @@
+     */
+    boolean isContainerNative() {
+        return false
+    }
```
Does `wr` manage the container execution on its own? In that case, it should be `true`.

sb10 replied:
As in doing something if the user wishes to run a docker container? No, wr doesn't yet have native container support. I'm hoping BashWrapperBuilder will put something appropriate in the wrapper script if this boolean returns false.

pditommaso replied:
Then it's fine and yes, BashWrapperBuilder should manage the container as usual.

pditommaso commented on modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy:
```diff
@@ -0,0 +1,664 @@
+    final List<String> COMPLETE_STATUSES = ['complete', 'buried', 'deleted']
+
+    final List<String> STARTED_STATUSES = ['delayed', 'reserved', 'running'] + COMPLETE_STATUSES
+
```
Makes sense to declare these fields also `public static`

sb10 replied:
I made that change, thanks.

pditommaso commented on modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy:
```diff
@@ -0,0 +1,664 @@
+    static private String endpoint
+	static private String cacertPath
+	static private WrRestApi client
+
```
It looks there's a problem with tab/space indentation. Please use 4 spaces in place of tabs. More details [here](https://github.com/nextflow-io/nextflow/blob/master/CONTRIBUTING.md).

sb10 replied:
Should now be fixed

sb10 commented:
After I `make compile` I end up with:

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   docs/conf.py
	modified:   modules/nextflow/src/main/groovy/nextflow/Const.groovy
	modified:   nextflow
```

I did not modify these files myself,
Is this expected? Is there a way to avoid these modifications being reported by git? And to avoid conflicts with them etc?

pditommaso commented:
The `compile` automatically changes the timestamp and build number, but those changes should not be commited.

sb10 commented:
Superseded by #1148
